plugins/out_s3/s3.c

```diff
@@ -1490,14 +1564,14 @@ static struct multipart_upload *get_upload(struct flb_s3 *ctx,
 
     mk_list_foreach_safe(head, tmp, &ctx->uploads) {
         tmp_upload = mk_list_entry(head, struct multipart_upload, _head);
-
         if (tmp_upload->upload_state == MULTIPART_UPLOAD_STATE_COMPLETE_IN_PROGRESS) {
             continue;
         }
-        if (tmp_upload->upload_errors >= MAX_UPLOAD_ERRORS) {
+        if (tmp_upload->upload_errors > ctx->ins->retry_limit) {
             tmp_upload->upload_state = MULTIPART_UPLOAD_STATE_COMPLETE_IN_PROGRESS;
-            flb_plg_error(ctx->ins, "Upload for %s has reached max upload errors",
-                          tmp_upload->s3_key);
+            flb_plg_error(ctx->ins, "Multipart Upload for %s has reached max "
+                          "s3:UploadPart errors, plugin will try to complete this upload "
+                          "to prevent loss of already uploaded data", tmp_upload->s3_key);
```

Comment:
Remember in my note that I mentioned that we need to check if any data has been uploaded for the multipart upload- if no data was uploaded I think we can just delete the upload and remove it from list.

Reply:
this comment still applies?

Reply:
Hi?